opennms-webapp/src/main/webapp/admin/sched-outages/editoutage.jsp

```diff
@@ -331,285 +332,288 @@ Could not find an outage to edit because no outage name parameter was specified
-							continue;
-						} catch (IndexOutOfBoundsException ioob) {
-							//Ignore - it's already removed
-							continue;
-						}
-						break;
-					} else if (paramName.startsWith("deleteInterface")) {
-						String indexStr = paramName.substring("deleteInterface".length(), paramName.indexOf("."));
-						try {
-							int index = WebSecurityUtils.safeParseInt(indexStr);
-							theOutage.removeInterface(theOutage.getInterfaces().get(index));
-						} catch (NumberFormatException e) {
-							//Ignore - nothing we can do
-							continue;
-						} catch (IndexOutOfBoundsException ioob) {
-							//Ignore - it's already removed
-							continue;
-						}
-						break;
-					} else if (paramName.startsWith("deleteTime")) {
-						String indexStr = paramName.substring("deleteTime".length(), paramName.indexOf("."));
-						try {
-							int index = WebSecurityUtils.safeParseInt(indexStr);
-							theOutage.removeTime(theOutage.getTimes().get(index));
-						} catch (NumberFormatException e) {
-							//Ignore - nothing we can do
-							continue;
+					//Look for deleteNode or deleteInterface or deleteTime prefix
+					List<String> paramList = Collections.list(request.getParameterNames());
+					for (String paramName : paramList) {
+						if (paramName.startsWith("deleteNode")) {
+							String indexStr = paramName.substring("deleteNode".length(), paramName.indexOf("."));
+							try {
+								int index = WebSecurityUtils.safeParseInt(indexStr);
+								theOutage.removeNode(theOutage.getNodes().get(index));
+							} catch (NumberFormatException e) {
+								//Ignore - nothing we can do
+								continue;
+							} catch (IndexOutOfBoundsException ioob) {
+								//Ignore - it's already removed
+								continue;
+							}
+							break;
+						} else if (paramName.startsWith("deleteInterface")) {
+							String indexStr = paramName.substring("deleteInterface".length(), paramName.indexOf("."));
+							try {
+								int index = WebSecurityUtils.safeParseInt(indexStr);
+								theOutage.removeInterface(theOutage.getInterfaces().get(index));
+							} catch (NumberFormatException e) {
+								//Ignore - nothing we can do
+								continue;
+							} catch (IndexOutOfBoundsException ioob) {
+								//Ignore - it's already removed
+								continue;
+							}
+							break;
+						} else if (paramName.startsWith("deleteTime")) {
+							String indexStr = paramName.substring("deleteTime".length(), paramName.indexOf("."));
+							try {
+								int index = WebSecurityUtils.safeParseInt(indexStr);
+								theOutage.removeTime(theOutage.getTimes().get(index));
+							} catch (NumberFormatException e) {
+								//Ignore - nothing we can do
+								continue;
+							}
+							break;
 						}
-						break;
 					}
 				}
-			}
 
-		} finally {
-			pollOutagesDao.getWriteLock().unlock();
-		}
+			} finally {
+				pollOutagesDao.getWriteLock().unlock();
+			}
 
-	} //end if form submission
+			response.sendRedirect("editoutage.jsp");
```

Comment:
If I'm reading this correctly, the only actual changes here are this line and the addition of the outer 'if method is POST', correct?